packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -281,3 +281,187 @@ def thread_request():
+    namespace. You can read more about the motivation for this CNI plugin and type
+    of problems it solves in this design doc:
+
+    https://docs.google.com/document/d/1xxvkFknC56hF-EcDmZ9tzKsGiZdGKBUPfrPKYs85j1k/edit?usp=sharing
+
+    In order to test `dcos-l4lb` CNI plugin we emulate a virtual network that lacks
+    routes for spartan interface and minuteman VIPs. In this test, we first install
+    a virtual network called `spartan-net` on one of the agents. The `spartan-net`
+    is a CNI network that is a simple BRIDGE network with the caveat that it
+    doesn't have any default routes. `spartan-net` has routes only for the agent
+    network and the DC/OS overlay (9.x.x.x/8) In other words it doesn't have any
+    routes towards the spartan-interfaces or minuteman VIPs.
+
+    We then run a server (our python ping-pong server) on the DC/OS overlay.
+    Finally to test that the `dcos-l4lb` plugin, which is also part of
+    `spartan-net` is able to inject the Minuteman and Spartan services into the
+    contianer's netns, we start a client on the `spartan-net` and try to `curl` the
+    `ping-pong` server using its VIP. Without the Minuteman and Spartan services
+    injected in the container's netns the expectation would be that this `curl`
+    would fail, with a successful `curl` execution on the VIP allowing the
+    test-case to PASS.
+    '''
+    # Get the first agent. We will run our apps on this specific agent.
+    log.debug("Retrieving agent network")
+    agent_network = None
+    try:
+        agent_network = ipaddress.IPv4Interface(dcos_api_session.slave_list[1] + '/8').network
+    except Exception as ex:
+        raise AssertionError(
+            "Couldn't retrieve the agent network':{}".format(agent_network)) from ex
+
+    log.info("Agent network is:{}".format(agent_network))
+
+    '''
+    Test the l4lb CNI plugin
+    * create an app that will install a CNI configuration for `spartan-net` network on all the agents.
+    * launch a server container on an agent on the `spartan-net` network.
+    * launch a client on the `spartan-net` network on the same agent as the server was launched.
+    * make the client get information from the server using the VIP.
+    '''
+    spartan_net = {
+        'cniVersion': '0.2.0',
+        'name': 'spartan-net',
+        'type': 'dcos-l4lb',
+        'delegate': {
+            'type': 'mesos-cni-port-mapper',
+            'excludeDevices': ['sprt-cni0'],
+            'chain': 'spartan-net',
+            'delegate': {
+                'type': 'bridge',
+                'bridge': 'sprt-cni0',
+                'ipMasq': True,
+                'isGateway': True,
+                'ipam': {
+                    'type': 'host-local',
+                    'subnet': '192.168.250.0/24',
+                    'routes': [
+                     # Reachability to the default `dcos` overlay subnet.
+                     {'dst': '9.0.0.0/8'},
```

Comment:
I guess we don't need reachability to overlay as client would be accessing the server via VIP

Reply:
Good point.

Reply:
My bad !! We actually need the 9.0.0.0/8 route in spartan-net since the IPVS points to an address from this subnet, since we are launching the server on the overlay.